Detectors/TPC/simulation/include/TPCSimulation/DigitGlobalPad.h

```diff
@@ -82,31 +90,36 @@ class DigitGlobalPad
   bool compareMClabels(const MCCompLabel& label1, const MCCompLabel& label2) const;
 
   float mChargePad;                                  ///< Total accumulated charge on that GlobalPad for a given time bin
-  std::vector<std::pair<MCCompLabel, int>> mMClabel; ///< vector to accumulate the MC labels
+  int mID = -1;                                      ///< ID of this digit to refer into labels (-1 means not initialized)
 };
 
-inline DigitGlobalPad::DigitGlobalPad() : mChargePad(0.), mMClabel() {}
+inline DigitGlobalPad::DigitGlobalPad() : mChargePad(0.) {}
 
-inline void DigitGlobalPad::addDigit(const MCCompLabel& label, float signal)
+inline void DigitGlobalPad::addDigit(const MCCompLabel& label, float signal,
+                                     o2::dataformats::LabelContainer<std::pair<MCCompLabel, int>, false>& labels)
 {
   bool isKnown = false;
-  // MCCompLabel tempLabel(trackID, eventID);
-  for (auto& mcLabel : mMClabel) {
+  auto view = labels.getLabels(mID);
+  int size = view.size();
```

Comment:
`size` is not used.

Reply:
thanks. removed. Will be merged thereafter.